soroban-sdk/src/env.rs

```diff
@@ -522,6 +488,21 @@ impl Env {
         }
         contract_id
     }
+
+    pub fn verify_account_authorization(
```

Comment:
When would a contract need to manually verify account authorization, vs relying on the system to verify an invocation or not?

Reply:
This is a testutils-only function (which is a bit tricky to notice from the diff). The idea is that tests have auth in the recording mode. Then it's possible to write the tests without worrying about auth at all. But, of course, it's important to cover the authorizations of your own contract at least once, hence we provide a function like this one that allows to check that an account would need to authorize something. See the example tests for the examples of using this function.